nep-xx.mediawiki

```diff
@@ -0,0 +1,101 @@
+<pre>
+  NEP:
+  Title: Contract Basic Methods Guideline
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>
+  Type: Informational
+  Status: Draft
+  Created: 2021-04-19
+  Replaces: 0
+</pre>
+
+
+
+
+
+
+==Abstract==
+
+This proposal outlines a basic method guideline for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract initial deploy, update and destroy.
+
+==Motivation==
+
+As the NEO blockchain scales, Smart Contract Initial Deploy, Update and Destroy will become increasingly important.  Without guideline for <code>_deploy</code>, <code>update</code> and <code>destroy</code> method, systems will be required to maintain a unique API for each contract, regardless of their similarity to other contracts which makes application development very inconvenient. This guideline can lead most contracts to implement the same basic methods with the same types of parameters but still accept diversity.
```

Comment:
I'm not entirely sure we should put `_deploy` there on the same line as `update` or `destroy` because it's a system-defined method, not a convention (it's even briefly mentioned on https://docs.neo.org/docs/en-us/reference/scapi/framework/native/ContractManagement/Deploy.html). But maybe this NEP will make it somewhat more visible, probably not all developers even know about it.

Reply:
Done